### ISSUE TITLE
locateOnScreen not working on Big Sur

### ISSUE THREAD

**Issue by oidualc** [asweigart/pyautogui#578]
After upgrading to macOS Big Sur, the function locateOnScreen is unable to find any image.

**Steps to reproduce:**

- Make sure "myimage.png" is visible on screen
- run this code `pyautogui.locateOnScreen("myimage.png")`

**Expected result:**

The position is returned by the function.

**Actual result:**

`None` is returned by the function.

**Comment by oidualc** [comment]
To workaround this, use the `confidence` argument (requires OpenCV) -> `pyautogui.locateOnScreen("myimage.png", confidence=0.9)`

**Comment by rez1-inf** [comment]
`pyautogui.locateOnScreen("image.png")` works for me in MacOS, but the coordinates are about a hundred pixel off for X and Y.

I looked at the source code and it seems `pyautogui` uses `pyscreeze` by the same author to locate stuff on screen. So this issue might be within `pyscreeze`

@oidualc were you able to get `locateOnScreen()` to work on mac? Also, were the coordinates returned correct for you?

**Comment by oidualc** [comment]
No, as mentioned in this bug `pyautogui.locateOnScreen("myimage.png")` does not work at all since I upgraded to Big Sur, it works only with the confidence parameter.

**Comment by rez1-inf** [comment]
> No, as mentioned in this bug `pyautogui.locateOnScreen("myimage.png")` does not work at all since I upgraded to Big Sur, it works only with the confidence parameter.

that is correct. so to overcome this issue we can use `locateCenterOnScreen("img.png")` and divide each coordinate by 2.

**Comment by rez1-inf** [comment]
it works for me without `confidence` parameter, but the coordinates are off by a factor of 2.

**Comment by damanc7** [comment]
I'm having the same issue.  Without a confidence  parameter I get the result `None` is returned and with a confidence parameter the image is found but with very off coordinates.

**Comment by rez1-inf** [comment]
@damanc7 you can solve that issue by dividing the coordinates by the device pixel ratio.. it is explained here https://github.com/asweigart/pyautogui/issues/589#issuecomment-876287189

**Comment by cj81499** [comment]
@rez1-inf I think the point is that this is really something the library ought to handle for the user. Correcting for display scaling is not something that I want to spend any time thinking about as a developer trying to throw some GUI automation together. While the workaround you linked is useful, this is still a BIG usability issue imo.

**Comment by rez1-inf** [comment]
@cj81499 i agree.. this workaround could easily be implemented as a fix.. but PyAutoGUI uses PyScreeze by the same author to do most of the work.. and nobody seems to be updating that anymore.. I'll try implementing the fix there and do a pull request sometime soon.. let's see if that'll work..
